### PR TITLE
Fix rm-portal lost advanced params when submit with hiding advanced param

### PR DESCRIPTION
**Pull request by tao-xinxiu**
Bug description: the edition of advanced parameters value are not correctly saved then submitted when saveNodeSource. In addition, if once shown the advanced parameters, then submit to saveNodeSource with unchecked Advanced Configuration will get null exception.

Bug reason: hiddenItems are not correctly updated when advanced parameters changed.

### CODE REVIEW

**Comment by tao-xinxiu** [comment]
retest this please

**Comment by tao-xinxiu** [comment]
retest this please

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java [hunk 1]
```diff
@@ -106,8 +105,14 @@
 
     private static final String HIDDEN_INFRA = "hidden-infra";
 
+    // mark whether the form item correspond an important field (i.e., whether it should be shown when hiding advanced configuration)
```
Comment (MaelAudren) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java:108]: Maybe `required field` (vs optional) would be more relevant than important

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java [hunk 2]
```diff
@@ -601,6 +612,29 @@ private void addPolicyPluginValuesToAllFormItems(Collection<PluginDescriptor> al
         this.fullPolicyValueMap = selectItemValues;
     }
 
+    // populate hiddenItems from current form items value.
```
Comment (MaelAudren) [inline rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java:615]: so here you could have optional items

File: rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java (hunk 1 above)
Comment (fviale) [reply rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/NodeSourceWindow.java:108]: an important field is not necessarily required. It simply means that it's a field cpmmonly modified, but not all important fields need to have a value

**Pull request merged**